File: data/code_pairs/fortran/737.f90
SUBROUTINE FIND(ibuffer, ib, nini, ntot, ip, iat)
    IMPLICIT NONE
    INTEGER, INTENT(IN) :: ibuffer(*), ib, nini, ntot, ip
    INTEGER, INTENT(OUT) :: iat
    INTEGER :: ipx, i

    ipx = 0
    iat = 0
    DO i = nini+1, ntot
        IF (ibuffer(i) == ib) THEN
            ipx = ipx + 1
            IF (ip == ipx) THEN
                iat = i
                RETURN
            END IF
        END IF
    END DO
END SUBROUTINE FIND

File: data/code_pairs/cpp/737.cpp
#include <vector>

void FIND(const std::vector<int>& ibuffer, int ib, int nini, int ntot, int ip, int& iat) {
    int ipx = 0;
    // Note: Adjusting for C++ 0-based indexing in the loop
    for (iat = nini; iat < ntot; ++iat) {
        if (ibuffer[iat - 1] == ib) { // Adjusted index for 0-based array
            ++ipx;
            if (ip == ipx) {
                return; // Exiting the function if the condition is met
            }
        }
    }
    iat = -1; // Indicates not found or condition not met
}